src/main/java/com/openshift/internal/restclient/model/PersistentVolume.java

```diff
@@ -0,0 +1,276 @@
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.openshift.internal.restclient.model;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+
+import com.openshift.internal.restclient.model.volume.NFSVolume;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.model.volume.IPersistentVolume;
+import com.openshift.restclient.model.volume.VolumeType;
+import com.openshift.restclient.utils.MemoryUnit;
+
+
+public class PersistentVolume extends KubernetesResource implements IPersistentVolume {
+
+	private static final String PV_ACCESS_MODES = "spec.accessModes";
+	private static final String PV_CAPACITY = "spec.capacity.storage";
+	private static final String PV_RECLAIM_POLICY = "spec.persistentVolumeReclaimPolicy";
+	private static final String PV_NFS = "spec.nfs";
+	private static final String PV_HOST_PATH = "spec.hostPath";
+	private static final String PV_SPEC = "spec";
+	private static final String SERVER = "server";
+	private static final String PATH = "path";
+	private static final String READ_ONLY = "readOnly";
+
+	public PersistentVolume(ModelNode node, IClient client, Map<String, String[]> propertyKeys) {
+		super(node, client, propertyKeys);
+	}
+
+	/**
+	 * @param unit the designated unit. One of "Ki", "Mi", "Gi", "Ti", "Pi", "Ei".
+	 * @return 0, if conversion not possible. Otherwise the capacity in given units.
+	 * @see {@link PersistentVolume#convert(String, MemoryUnit)}
+	 * @throws IllegalArgumentException if not supported {@code unit} given.
+	 */
+	@Override
+	public long getCapacity(String unit) {
+		String capacity = asString(PV_CAPACITY);
+		return convert(capacity, MemoryUnit.valueOf(unit));
+	}
+
+	/**
+	 * @param unit the designated unit. One of {@link MemoryUnit}'s Ki, Mi, Gi, Ti, Pi, Ei.
+	 * @return 0, if conversion not possible. Otherwise the capacity in given units.
+	 * @see {@link PersistentVolume#convert(String, MemoryUnit)}
+	 * @throws IllegalArgumentException if not supported {@code unit} given.
+	 */
+	@Override
+	public long getCapacity(MemoryUnit unit) {
+		String capacity = asString(PV_CAPACITY);
+		return convert(capacity, unit);
+	}
+
+	/**
+	 * @return the capacity in bytes.
+	 */
+	@Override
+	public long getCapacity() {
+		String capacity = asString(PV_CAPACITY);
+		return Math.multiplyExact(convert(capacity, MemoryUnit.Ki), 1024L);
+	}
+
+	/**
+	 * Sets the capacity. There is no conversion between units.
+	 * @param capacity the capacity
+	 * @param unit the unit
+	 */
+	@Override
+	public void setCapacity(long capacity, MemoryUnit unit) {
+		get(PV_CAPACITY).set(capacity + unit.name());
```

Comment:
There are convenience methods as part of KubernetesResource that provide this logic, set(path,value)